server/channels/store/storetest/channel_store_categories.go

```diff
@@ -2413,3 +2423,251 @@ func testSidebarCategoryDeadlock(t *testing.T, rctx request.CTX, ss store.Store)
+					},
+					{
+						SidebarCategory: customCategories[0].SidebarCategory,
+						Channels:        []string{},
+					},
+				}
+			case 2:
+				// Move a DM channel to a custom category
+				updatedCategories = []*model.SidebarCategoryWithChannels{
+					{
+						SidebarCategory: initialCategories.Categories[2].SidebarCategory, // DMs category
+						Channels:        []string{},
+					},
+					{
+						SidebarCategory: customCategories[1].SidebarCategory,
+						Channels:        []string{<-dmChannelChan, <-channelChan, <-channelChan},
+					},
+				}
+			case 3:
+				// Move a DM channel back from to a custom category
+				updatedCategories = []*model.SidebarCategoryWithChannels{
+					{
+						SidebarCategory: initialCategories.Categories[2].SidebarCategory, // DMs category
+						Channels:        []string{<-dmChannelChan},
+					},
+					{
+						SidebarCategory: customCategories[1].SidebarCategory,
+						Channels:        []string{<-channelChan, <-channelChan},
+					},
+				}
+			case 4:
+				// Move a channel between custom categories
+				updatedCategories = []*model.SidebarCategoryWithChannels{
+					{
+						SidebarCategory: customCategories[0].SidebarCategory,
+						Channels:        []string{<-channelChan, <-channelChan, <-channelChan},
+					},
+					{
+						SidebarCategory: customCategories[1].SidebarCategory,
+						Channels:        []string{<-channelChan},
+					},
+				}
+			case 5:
+				// Move a channel back between custom categories
+				updatedCategories = []*model.SidebarCategoryWithChannels{
+					{
+						SidebarCategory: customCategories[0].SidebarCategory,
+						Channels:        []string{<-channelChan},
+					},
+					{
+						SidebarCategory: customCategories[1].SidebarCategory,
+						Channels:        []string{<-channelChan, <-channelChan, <-channelChan},
+					},
+				}
+			case 6:
+				// Add to favorites category (triggers preference updates)
+				updatedCategories = []*model.SidebarCategoryWithChannels{
+					{
+						SidebarCategory: initialCategories.Categories[0].SidebarCategory, // Favorites category
+						Channels:        []string{<-channelChan, <-dmChannelChan},
+					},
+				}
+			case 7:
+				// Update category properties
+				customCatIndex := rand.Intn(2)
+				updatedCategories = []*model.SidebarCategoryWithChannels{
+					{
+						SidebarCategory: model.SidebarCategory{
+							Id:          customCategories[customCatIndex].Id,
+							DisplayName: fmt.Sprintf("Updated Name %d", iteration),
+							Sorting:     model.SidebarCategorySortRecent,
+							Muted:       iteration%2 == 0,
+						},
+						Channels: customCategories[customCatIndex].Channels,
+					},
+				}
+			}
+
+			_, _, err := ss.Channel().UpdateSidebarCategories(userID, team.Id, updatedCategories)
+			if err != nil {
+				for _, cat := range updatedCategories {
+					t.Logf("[iteration %d] sidebar category - %v, channels: %s \n", iteration, cat.SidebarCategory, cat.Channels)
```

Comment:
Let's remove this as well. This was a line I added to understand the deadlock. This leads to too many log lines.